sdk/src/server-api/sc/framework/plugins/RoundBasedGameInstance.kt

```diff
@@ -0,0 +1,269 @@
+package sc.framework.plugins
+
+import com.thoughtworks.xstream.annotations.XStreamAsAttribute
+import com.thoughtworks.xstream.annotations.XStreamImplicit
+import com.thoughtworks.xstream.annotations.XStreamOmitField
+import org.slf4j.LoggerFactory
+import sc.api.plugins.IGameInstance
+import sc.api.plugins.IGameState
+import sc.api.plugins.exceptions.GameLogicException
+import sc.api.plugins.host.IGameListener
+import sc.protocol.responses.ProtocolErrorMessage
+import sc.protocol.responses.ProtocolMessage
+import sc.shared.InvalidMoveException
+import sc.shared.PlayerScore
+import sc.shared.ScoreCause
+import sc.shared.WinCondition
+import java.util.*
+import kotlin.jvm.Throws
+
+abstract class RoundBasedGameInstance<P : Player>(@XStreamOmitField override val pluginUUID: String) : IGameInstance {
+    companion object {
+        val logger = LoggerFactory.getLogger(RoundBasedGameInstance::class.java)
+    }
+
+    protected var activePlayer: P? = null
+
+    @XStreamAsAttribute
+    var turn = 0
+        private set
+    open val round: Int
+        get() = turn / 2
+
+    @XStreamOmitField
+    private var paused = Optional.empty<Int>()
```

Comment:
`Optional` is not supposed to ever be stored. This should simply be a nullable Int.

Reply:
why using `Optional` at all, then?

Reply:
188ae55d

Reply:
because Java doesn't handle null well ;)

Reply:
> However, with the current state the GUI fails to recognise Game Over

I fixed that error. As I see it, everything is working now